src/main/java/com/imsweb/validation/functions/MetafileContextFunctions.java

```diff
@@ -975,6 +975,88 @@ else if (index instanceof Set)
 
     /**
      * Special genedit method. Internal use only.
+     */
+    public boolean GEN_SQLLOOKUP(List<List<Object>> table, List<Object> columnNames, Object value) {
+        if (columnNames.isEmpty())
+            return true;
+
+        String val = GEN_TO_STRING(value);
+
+        int[] colNumbers = new int[columnNames.size()];
+        for (int i = 0; i < colNumbers.length; i++) {
+            colNumbers[i] = table.get(0).indexOf(columnNames.get(i));
+            if (colNumbers[i] == -1)
+                return true;
+        }
+
+        for (int r = 1; r < table.size(); r++) {
+            int index = 0;
+            boolean match = false;
+            StringBuilder rowVal = new StringBuilder("");
+            for (int c : colNumbers)
+                rowVal.append(GEN_TO_STRING(table.get(r).get(c)));
+
+            if (val.equals(rowVal.toString()))
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Special genedit method. Internal use only.
+     */
+    public boolean GEN_SQLRANGELOOKUP(List<List<Object>> table, List<Object> columnNames, Object value, Map<Integer, char[]> tableVars) {
+        if (columnNames.isEmpty())
+            return true;
+
+        String val = GEN_TO_STRING(value);
+
+        int[] colNumbers = new int[columnNames.size()];
+        for (int i = 0; i < colNumbers.length; i++) {
+            colNumbers[i] = table.get(0).indexOf(columnNames.get(i));
+            if (colNumbers[i] == -1)
+                return false;
+        }
+
+        int indexOfLargest = -1;
+        String largest = "";
+        for (int r = 1; r < table.size(); r++) {
+            StringBuilder rowVal = new StringBuilder("");
+            for (int c : colNumbers)
+                rowVal.append(GEN_TO_STRING(table.get(r).get(c)));
+            if (StringUtils.isNumeric(val) && StringUtils.isNumeric(rowVal)) {
```

Comment:
String.compare("2", "10") returns 2 > 10. Their code returns 2 < 10. In the case where the values are alphanumeric, it uses String compare (or something similar), so "A2" > "A10" for them, but for straight numbers we need separate logic.